src/resmom/mom_comm.c

```diff
@@ -217,6 +219,41 @@ pbs_jobnodevoid_t	job_free_node = NULL;
 
 eventent * event_dup(eventent *ep, job *pjob, hnodent *pnode);
 
+/**
+ * @brief
+ *	Write to fd and check if enough was written
+ *
+ * @param[in]   fd - file descriptor
+ * @param[in]   write_ptr - pointer to data to be written
+ * @param[in]   size - size of data which should be written
+ *
+ * @return   Error code
+ * @retval   0 Success
+ * @retval  -1 Failure
+ *
+ */
+int
+ensure_write(int fd, char *write_ptr, int size)
+{
+	int i = 0;
+	while ((i = write(fd, write_ptr, size)) != size) {
+		if ((i < 0) && (errno == EINTR)) {	/* retry the write */
+			if (lseek(fd, (off_t)0, SEEK_SET) < 0) {
```

Comment:
Doing lseek 0 here can lead to corrupted data in .TK file...
For example, if ensure_write is called for `obit->oe_u.oe_tm.oe_node` (4th write in file), and if we do lseek 0 then will loose data for `ptask->ti_qs`, `pobit->oe_type`, `pobit->oe_u.oe_tm.oe_fd`